### PR TITLE
Added some error handlers and robustness checks for setup script Fixes #32

### LINKED ISSUE #32: Setup script missing some robustness features
Context:
When using iconarray in an existing project the setup-script can fail and return an error. Error-handling is not very robust in the current version of the env/setup-conda-env.sh script.

For example:
When copying the env/siteconfig.py to the cartopy source folder, the script fails if multiple python versions are installed (especially not 3.9.7).

Solution:
Catch some error messages, write some robust code to allow for varying user setups.

### PR DESCRIPTION
**Pull request by sadamov**
Hi iconarray-team,

Big fan of the project and first time contributor (but longer time user).

Here some reasoning for the stuff I changed in the setup file:
- `$HOST` is sometimes not defined in non-login shells, using `$(hostname -s)` is more robust. This can be important for VSCode users for example
- Made sure that if module loading fails, the script aborts and returns an error message
- The `~`-sign has to be escaped when working in zsh, in bash both `\~` and `~` work
- My default formatter changed `` to `$()` syntax, which I think is clearer to read and less prone to quotation mark errors
- Copying the cartopy script failed because of non-matching Python versions, the Python version retrieval is now automatic
- Added some error messages

### CODE REVIEW

File: env/setup-conda-env.sh [hunk 1]
```diff
@@ -42,7 +55,12 @@ fi
...
 # ---- required for cartopy ------
 
-echo 'Make sure to deactivate your environment completely before reactivating (i.e., you should not be in a base environment). To make sure, you can run conda deactivate twice'
-echo 'Enable cartopy to modify cartopy.config by placing siteconfig.py in cartopy package'
-vpython=$(ls --color=never -d $CONDA_PREFIX/lib/python*);
-cp env/siteconfig.py ${vpython}/site-packages/cartopy
+python_version=$(python --version | tr '[:upper:]' '[:lower:]' | sed 's/ //g' | sed 's/\.[^.]*$//')
+vpython=$(ls --color=never -d $CONDA_PREFIX/lib/$python_version)
+if cp env/siteconfig.py ${vpython}/site-packages/cartopy; then
+    echo -e 'The setup script completed successfully! \n' \
```
Comment (victoria-cherkas) [inline env/setup-conda-env.sh:61]: I think we should decouple the logging of "setup script completed successfully" from this cartopy configuration. ie log error and exit if siteconfig.py can't be copied, and separately log successful script completion.

**Review by victoria-cherkas — COMMENTED** [review]
Thanks a lot for this PR! Lots of improvements that were much needed. Just a couple of comments and small changes requested. 

I also think it would be useful to log the results of conda env config vars list  (just before script says it has successfully completed). Happy to do this and some of the other bits myself , just let me know :)

File: env/setup-conda-env.sh [hunk 2]
```diff
@@ -1,24 +1,37 @@
 #!/bin/bash
 
-if [[ $HOST == *'tsa'* ]]; then
+if [[ $(hostname -s) == *'tsa'* ]]; then
     echo 'Setting GRIB_DEFINITION_PATH for cfgrib engine'
-    module load python
+    errormessage=$(module load python)
+    if [ -z "$errormessage" ]; then
```
[the remark this answers is not in the document]
Comment (sadamov) [reply env/setup-conda-env.sh:6]: I think the same syntax we use below should be better. As it captures errors during execution as well as missing commands.

File: env/setup-conda-env.sh [hunk 3]
```diff
@@ -1,24 +1,37 @@
...
+        echo 'EasyBuild loaded the python module successfully.'
+    else
+        echo 'EasyBuild was not able to load the python module.'
+        return $1
+    fi
     source /project/g110/spack/user/admin-tsa/spack/share/spack/setup-env.sh
 
-    cosmo_eccodes=`spack find --format "{prefix}" cosmo-eccodes-definitions@2.19.0.7%gcc | head -n1`
-    eccodes=`spack find --format "{prefix}" eccodes@2.19.0%gcc ~aec | head -n1`
+    cosmo_eccodes=$(spack find --format "{prefix}" cosmo-eccodes-definitions@2.19.0.7%gcc | head -n1)
```
[the remark this answers is not in the document]
Comment (sadamov) [reply env/setup-conda-env.sh:14]: Yeah the error message is not very clear. I tried to add some more information.

File: env/setup-conda-env.sh (hunk 1 above)
Comment (sadamov) [reply env/setup-conda-env.sh:61]: I agree, I captured the remaining possible errors above. Now, we can be sure that if the script ran through it was successfull and is allowed to print a happy message.

**Review by victoria-cherkas — APPROVED** [review]

**Comment by sadamov** [comment]
Ready to merge from my side :partying_face:

**Review by victoria-cherkas — APPROVED** [review]

**Pull request merged**